packages/react-native-sdk/src/components/PermissionsNotification.tsx

```diff
@@ -0,0 +1,109 @@
+import { useCall, useHasPermissions } from '@stream-io/video-react-bindings';
+import {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Alert, Pressable } from 'react-native';
+
+export type PermissionNotificationProps = PropsWithChildren<{
+  /**
+   * The permission to check for.
+   */
+  permission: OwnCapability;
+
+  /**
+   * The message to display in the notification once
+   * the requested permission is granted.
+   */
+  messageApproved: string;
+
+  /**
+   * The message to display in the notification once a permission
+   * is revoked.
+   */
+  messageRevoked: string;
+
+  /**
+   * The message to display in the notification while
+   * the requested permission is awaiting approval.
+   */
+  messageAwaitingApproval: string;
+}>;
+
+export const PermissionNotification = (props: PermissionNotificationProps) => {
+  const {
+    permission,
+    messageApproved,
+    messageRevoked,
+    messageAwaitingApproval,
+    children,
+  } = props;
+  const hasPermission = useHasPermissions(permission);
+  const previousHasPermission = useRef(hasPermission);
+
+  const [isAwaitingApproval, setIsAwaitingApproval] = useState(false);
+  const call = useCall();
+
+  const showGrantedNotification = useCallback(() => {
+    Alert.alert(messageApproved);
+  }, [messageApproved]);
+
+  const showRevokedNotification = useCallback(() => {
+    Alert.alert(messageRevoked);
+  }, [messageRevoked]);
+
+  const showApprovalRequestNotification = useCallback(() => {
+    Alert.alert(messageAwaitingApproval);
+  }, [messageAwaitingApproval]);
+
+  useEffect(() => {
+    if (hasPermission) {
+      setIsAwaitingApproval(false);
+    }
+  }, [hasPermission]);
+
+  useEffect(() => {
+    if (hasPermission && !previousHasPermission.current) {
+      previousHasPermission.current = true;
+      showGrantedNotification();
+    } else if (!hasPermission && previousHasPermission.current) {
+      previousHasPermission.current = false;
+      showRevokedNotification();
+    } else if (!hasPermission && isAwaitingApproval) {
+      previousHasPermission.current = false;
+      showApprovalRequestNotification();
+    }
+  }, [
+    isAwaitingApproval,
+    hasPermission,
+    showGrantedNotification,
+    showRevokedNotification,
+    showApprovalRequestNotification,
+  ]);
+
+  const handleRequestPermission = useCallback(async () => {
+    if (call?.permissionsContext.canRequest(permission)) {
+      setIsAwaitingApproval(true);
+      await call
+        .requestPermissions({ permissions: [permission] })
+        .catch((error) => {
+          if (error instanceof AxiosError) {
+            console.log(
+              'RequestPermissions failed',
+              error.response?.data.message,
+            );
+          }
+        });
+    }
+  }, [call, permission]);
+
+  if (!hasPermission && !isAwaitingApproval) {
+    return <Pressable onPress={handleRequestPermission}>{children}</Pressable>;
+  }
```

Comment:
This will become a pressable that wraps a pressable :/ I'm confused

Reply:
Its not actually Pressable on Pressable, the logic works such that the upper pressable will be triggered always when the permissions are lacking for a given call control button.

Reply:
Anyways I have refactored the way it was written, hope it will make sense and be more readable now.